crates/miden-objects/src/batch/proven_batch.rs

```diff
@@ -0,0 +1,88 @@
+use alloc::{collections::BTreeMap, vec::Vec};
+
+use crate::{
+    account::AccountId,
+    batch::{BatchAccountUpdate, BatchId, BatchNoteTree},
+    block::BlockNumber,
+    transaction::{InputNoteCommitment, OutputNote},
+};
+
+/// A transaction batch with an execution proof.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ProvenBatch {
+    id: BatchId,
+    account_updates: BTreeMap<AccountId, BatchAccountUpdate>,
+    input_notes: Vec<InputNoteCommitment>,
```

Comment:
Should we use `InputNotes<InputNoteCommitment>` here instead of `Vec<InputNoteCommitment>`?